### PR TITLE
refactor: migrate Avatar to typescript

### PR DESCRIPTION
**Pull request by SumeetHaryani**
Add fix for issue #893  

#### Basic
- [ ] Used plop (`npm run plop`) to create a new component.
- [x] PR has description.
- [x] New component is functional and uses Hooks. 
- [x] Component defines [`PropTypes`](https://reactjs.org/docs/typechecking-with-proptypes.html).
#### Style
- [ ] Styles are added to `NewComponent.modules.scss` file inside of the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [x] Design is compatible with [Monday Design System](https://design.monday.com/).
#### Storybook
- [ ] Stories were added to `/src/NewComponent/__stories__/NewComponent.stories.js` file.
- [x] Stories include all flows of using the component.
#### Tests
- [x] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

### CODE REVIEW

File: src/components/Avatar/__tests__/avatar-snapshot-tests.jest.tsx [hunk 1]
```diff
@@ -3,7 +3,8 @@ import renderer from "react-test-renderer";
 import Avatar from "../Avatar";
 import { WhatsNew } from "../../Icon/Icons";
 
-jest.mock("../../Icon/Icon", () => ({ icon }) => <div data-testid="icon-mock">{icon}</div>);
+// TODO: Add type for icon
```
Comment (SumeetHaryani) [inline src/components/Avatar/__tests__/avatar-snapshot-tests.jest.tsx:6]: @orrgottlieb Could you please help me here with the type for the icon

Comment (orrgottlieb) [reply src/components/Avatar/__tests__/avatar-snapshot-tests.jest.tsx:6]: @SumeetHaryani you can define - 
`export type IconType =  string | React.FunctionComponent | null;` in types, and use this

File: src/components/Avatar/Avatar.tsx [hunk 2]
```diff
@@ -1,22 +1,60 @@
...
+  tooltipProps?: any;
+  ariaLabel?: string;
+  withoutTooltip?: boolean;
+  icon?: string | React.FunctionComponent<iconSubComponentProps> | null;
+  type?: AvatarTypes;
+  className?: string;
+  textClassName?: string;
+  backgroundColor?: BackgroundColors;
+  customBackgroundColor?: string;
+  role?: string;
```
Comment (orrgottlieb) [inline src/components/Avatar/Avatar.tsx:35]: `AriaRole` please

File: src/components/Avatar/Avatar.tsx [hunk 3]
```diff
@@ -1,22 +1,60 @@
...
+  isSquare?: boolean;
+  // Backward compatibility for props naming
+  isDisabled?: boolean;
+  square?: boolean;
+  topLeftBadgeProps?: AvatarBadgeProps;
+  topRightBadgeProps?: AvatarBadgeProps;
+  bottomLeftBadgeProps?: AvatarBadgeProps;
+  bottomRightBadgeProps?: AvatarBadgeProps;
+  withoutBorder?: boolean;
+  /**
```
Comment (orrgottlieb) [inline src/components/Avatar/Avatar.tsx:51]: change this comment or delete it please

File: src/components/Avatar/Avatar.tsx [hunk 4]
```diff
@@ -1,22 +1,60 @@
...
+   * (event) => void
+   */
+  onClick?: (event: React.MouseEvent) => void;
```
Comment (orrgottlieb) [inline src/components/Avatar/Avatar.tsx:54]: this could also be a keyboard event

File: src/components/Avatar/Avatar.tsx [hunk 5]
```diff
@@ -114,7 +152,7 @@ const Avatar = ({
   }, [size, topLeftBadgeProps, topRightBadgeProps, bottomLeftBadgeProps, bottomRightBadgeProps]);
 
   const clickHandler = useCallback(
-    event => {
+    (event: React.MouseEvent) => {
```
Comment (orrgottlieb) [inline src/components/Avatar/Avatar.tsx:155]: also could be a KeyboardEvent

File: src/components/Avatar/AvatarContent.tsx [hunk 6]
```diff
@@ -1,30 +1,46 @@
...
+    case AvatarTypes.IMG:
       return <img role={role} alt={ariaLabel} src={src} className={className} />;
-    case AVATAR_TYPES.ICON:
+    case AvatarTypes.ICON:
       return (
         <Icon
           icon={icon}
           aria-label={ariaLabel}
-          role={role}
+          // role={role}
```
Comment (orrgottlieb) [inline src/components/Avatar/AvatarContent.tsx:37]: why ?

File: src/components/Avatar/__tests__/avatar-snapshot-tests.jest.tsx [hunk 7]
```diff
@@ -22,12 +23,12 @@ describe("Avatar renders correctly", () => {
   });
 
   it("renders correctly with image", () => {
-    const tree = renderer.create(<Avatar img={IMG_SRC} />).toJSON();
+    const tree = renderer.create(<Avatar src={IMG_SRC} />).toJSON();
```
Comment (orrgottlieb) [inline src/components/Avatar/__tests__/avatar-snapshot-tests.jest.tsx:26]: nice catch!

**Review by orrgottlieb — CHANGES_REQUESTED** [review]
Hi @SumeetHaryani great great PR! thank you.
i have small comments there 🙏 

please tag me when you finish with the pr or you have any additional questions/comments ?

File: src/components/Avatar/__tests__/avatar-snapshot-tests.jest.tsx (hunk 1 above)
Comment (SumeetHaryani) [reply src/components/Avatar/__tests__/avatar-snapshot-tests.jest.tsx]: thanks @orrgottlieb

File: src/components/Avatar/Avatar.tsx (hunk 2 above)
Comment (SumeetHaryani) [reply src/components/Avatar/Avatar.tsx]: updated

File: src/components/Avatar/Avatar.tsx (hunk 3 above)
Comment (SumeetHaryani) [reply src/components/Avatar/Avatar.tsx]: updated

File: src/components/Avatar/Avatar.tsx (hunk 4 above)
Comment (SumeetHaryani) [reply src/components/Avatar/Avatar.tsx]: Added support for keyboard event

File: src/components/Avatar/Avatar.tsx (hunk 5 above)
Comment (SumeetHaryani) [reply src/components/Avatar/Avatar.tsx]: Added support for keyboard event

File: src/components/Avatar/AvatarContent.tsx (hunk 6 above)
Comment (SumeetHaryani) [reply src/components/Avatar/AvatarContent.tsx:37]: The Icon component doesn't have a role prop in it

**Review by orrgottlieb — APPROVED** [review]
@SumeetHaryani great addition and great contribution, thank you
approving, merging and adding the correct tag to the issue

p.s 
if you have any additional comment about the repo/library, please let me know 🙏

**Pull request merged**